pkg/datapath/linux/routing/migrate_test.go

```diff
@@ -0,0 +1,575 @@
+// mock is used. (2) is used for the downgrade test cases where the
+// GetMACByInterfaceNumber mock is used. These maps are used in their
+// respectives mocks to return the desired result data depending on the test.
+func setUpRoutingTable(c *C, ifindex, tableID, priority int) (map[string]int, map[string]string) {
+	devIfNum := make(map[string]int)
+	devMAC := make(map[string]string)
+
+	// Create n sets of a dummy interface, a route, and a rule.
+	//
+	// Each dummy interface has a /24 from the private range of 172.16.0.0/20.
+	//
+	// Each route will be a default route to the gateway IP of the interface's
+	// subnet.
+	//
+	// Each rule will be from the interface's subnet to all.
+	for i := 1; i <= n; i++ {
+		devName := fmt.Sprintf("gotestdummy%d", i)
+
+		gw := net.ParseIP(fmt.Sprintf("172.16.%d.1", i))
+		_, linkCIDR, err := net.ParseCIDR(fmt.Sprintf("172.16.%d.2/24", i))
+		c.Assert(err, IsNil)
+
+		linkIndex := ifindex + (i - 1)
+		newTableID := tableID + (i - 1)
+
+		dummyTmpl := &netlink.Dummy{
+			LinkAttrs: netlink.LinkAttrs{
+				Name:  devName,
+				Index: linkIndex,
+			},
+		}
+		c.Assert(netlink.LinkAdd(dummyTmpl), IsNil)
+		c.Assert(netlink.LinkSetUp(dummyTmpl), IsNil)
+		c.Assert(netlink.AddrAdd(dummyTmpl, &netlink.Addr{
+			IPNet: linkCIDR,
+		}), IsNil)
+		c.Assert(netlink.RouteAdd(&netlink.Route{
+			Dst:       &net.IPNet{IP: net.IPv4zero, Mask: net.CIDRMask(0, 32)},
+			Gw:        gw,
+			LinkIndex: dummyTmpl.Index,
+			Table:     newTableID,
+		}), IsNil)
+
+		// _, cidr, err := net.ParseCIDR("172.16.0.2/24")
+		// c.Assert(err, IsNil)
+		// c.Assert(netlink.RuleAdd(&netlink.Rule{
+		// 	// Src:      &net.IPNet{IP: net.ParseIP("172.16.0.2"), Mask: net.CIDRMask(24, 32)},
+		// 	Src: cidr,
+		// 	// Dst:      &net.IPNet{IP: net.IPv4zero, Mask: net.CIDRMask(0, 32)},
+		// 	Priority: linux_defaults.RulePriorityEgress,
+		// 	Table:    dummyTmpl.Index,
+		// }), IsNil)
+
+		// TODO(christarazi): Must shell out here due to netlink request (above)
+		// resulting in EINVAL. See https://github.com/cilium/cilium/issues/14383.
+		err = exec.Command("ip", "rule", "add",
+			"from", linkCIDR.String(),
+			"to", "all",
+			"table", fmt.Sprintf("%d", newTableID),
+			"priority", fmt.Sprintf("%d", priority)).Run()
+		c.Assert(err, IsNil)
+
+		// Return the MAC address of the dummy device, which acts as the ENI.
+		link, err := netlink.LinkByName(devName)
+		c.Assert(err, IsNil)
+
+		mac := link.Attrs().HardwareAddr.String()
+
+		// Arbitrarily use an offset of 1 as the interface number. It doesn't
+		// matter as long as we're consistent.
+		devIfNum[mac] = i
+		devMAC[devName] = mac
+	}
+
+	return devIfNum, devMAC
+}
+
+func findRulesByPriority(prio int) ([]netlink.Rule, error) {
+	rules, err := netlink.RuleList(netlink.FAMILY_V4)
+	if err != nil {
+		return nil, err
+	}
+
+	return filterRulesByPriority(rules, prio), nil
+}
+
+func (m *MigrateSuite) defaultNetlinkMock() {
```

Comment:
Not for this PR, but there are a number of projects that generate Go mocks, avoiding having to manually create them. One example is [gomock](https://github.com/golang/mock).